src/diff.rs

```diff
@@ -1,103 +1,252 @@
+                exclusive_left,
+                exclusive_right,
+                different_values,
+                common,
             }
         }
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use std::collections::HashMap;
+
+        use crate::diff::map::MapComparison;
+
+        #[test]
+        fn diff_empty_maps() {
+            let left: HashMap<&str, i32> = HashMap::from([]);
+            let right: HashMap<&str, i32> = HashMap::from([]);
+            let result = MapComparison::from_hash_maps(&left, &right);
+            assert!(result.common.is_empty());
+            assert!(result.exclusive_left.is_empty());
+            assert!(result.exclusive_right.is_empty());
+        }
+
+        #[test]
+        fn map_diff_left_exclusive() {
+            let left: HashMap<&str, i32> = HashMap::from([("123", 2)]);
+            let right: HashMap<&str, i32> = HashMap::from([]);
+            let result = MapComparison::from_hash_maps(&left, &right);
+            assert!(result.common.is_empty());
+            assert_eq!(result.exclusive_left, vec![(&"123", &2)]);
+            assert!(result.exclusive_right.is_empty());
+        }
+
+        #[test]
+        fn map_diff_right_exclusive() {
+            let left: HashMap<&str, i32> = HashMap::from([]);
+            let right: HashMap<&str, i32> = HashMap::from([("123", 2)]);
+            let result = MapComparison::from_hash_maps(&left, &right);
+            assert!(result.common.is_empty());
+            assert!(result.exclusive_left.is_empty());
+            assert_eq!(result.exclusive_right, vec![(&"123", &2)]);
+        }
 
-        MapComparison {
-            exclusive_left,
-            exclusive_right,
-            different_values,
-            common,
+        #[test]
+        fn map_diff_common() {
+            let left: HashMap<&str, i32> = HashMap::from([("123", 2)]);
+            let right: HashMap<&str, i32> = HashMap::from([("123", 2)]);
+            let result = MapComparison::from_hash_maps(&left, &right);
+            assert_eq!(result.common, vec![(&"123", &2)]);
+            assert!(result.exclusive_left.is_empty());
+            assert!(result.exclusive_right.is_empty());
         }
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use crate::diff::MapComparison;
-    use std::collections::HashMap;
+pub(crate) mod iter {
+    use std::fmt::Debug;
 
-    #[test]
-    fn diff_empty_maps() {
-        let left: HashMap<&str, i32> = HashMap::from([]);
-        let right: HashMap<&str, i32> = HashMap::from([]);
-        let result = MapComparison::from_hash_maps(&left, &right);
-        assert!(result.common.is_empty());
-        assert!(result.exclusive_left.is_empty());
-        assert!(result.exclusive_right.is_empty());
+    /// Differences between two Sequence-like structures.
+    pub(crate) struct SequenceComparison<T: PartialEq + Debug> {
+        pub(crate) order_preserved: bool,
+        pub(crate) exclusive_left: Vec<T>,
+        pub(crate) exclusive_right: Vec<T>,
```

Comment:
Could you rename left/right to missing/extras?  You implemented util methods like `are_same` unlike MapsComparison, which implicitly introduces the assumption of missing/extras.  The assumption should be clarified.

It would be nicer to rename `left`/`right` in `MapsComparison` for consistency, but this is optional.